Firmware/mmu2_protocol_logic.cpp

```diff
@@ -289,9 +313,8 @@ StepStatus Idle::Step() {
             return Processing;
         }
         break;
-    case State::QuerySent: { // check UART
-        auto expmsg = logic->ExpectingMessage(linkLayerTimeout);
-        if (expmsg != MessageReady)
+    case State::QuerySent: // check UART
+        if (auto expmsg = logic->ExpectingMessage(linkLayerTimeout); expmsg != MessageReady)
```

Comment:
Don't we need a curly brace {} around the case due to this variable initialization?

```
case State::QuerySent:
{
code
}
```

Reply:
I'm not sure if this is OK in C++17, but I'm probably thinking from C perspective.

Reply:
yes, this is C++17 and thus we don't need the { } anymore